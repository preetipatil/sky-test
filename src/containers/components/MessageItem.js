import React from 'react';

const mainDiv ={
    height: "150px",
}

const avatarDiv ={
    height: "100px",
    float:"left",
}

class MessageItem extends React.Component {

 constructor(){
    super();
    this.state = {
      emailTxt: "",
    };
 }

 onMouseEnter(email) {   ;
    this.setState({emailTxt: email})
 }

 onMouseLeave() {
    this.setState({emailTxt: ''})
 }

  render() {
    const { chatMessage, email, avatar, timestamp }  = this.props;
    
     return (
         <li >
            <div style={mainDiv}>
                <div style={avatarDiv}>
                    { avatar ? <img  src={avatar}  alt="" /> : " No Avatar "}
                </div>
                <div>
                    <p>DATE {new Date(timestamp).toLocaleString()} </p>
                    <p onMouseEnter={() => this.onMouseEnter(email)} onMouseLeave={() => this.onMouseLeave()}>TXT {chatMessage}  </p>
                    <span>{this.state.emailTxt}</span>
                </div>
            </div>
         </li>
     );
  }
}

export default MessageItem;