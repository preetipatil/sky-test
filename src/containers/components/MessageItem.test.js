import React from 'react';
import { shallow } from 'enzyme';
import MessageItem from './MessageItem';

describe('Message Items ', () => {
  it('Should render each message Item details', () => {
    const props = {
      chatMessage: 'Nulla tempus. Vivamus in felis eu sapien cursus vestibulum.',
      email: 'wgonzalezz@va.gov',
      avatar: 'http://dummyimage.com/100x100.png/ff4444/ffffff',
      timestamp: '2017-02-11T01:06:40Z',
    };

    const tree = shallow(<MessageItem {...props} />);
    expect(tree).toMatchSnapshot();
    expect(tree.find('li').length).toBe(1);
    //expect(tree.find('li').at(2)).toBe('');
  });
});
