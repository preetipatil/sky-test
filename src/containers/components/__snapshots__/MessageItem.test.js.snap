// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Items  Should render each message Item details 1`] = `
<li>
  <div
    style={
      Object {
        "height": "150px",
      }
    }
  >
    <div
      style={
        Object {
          "float": "left",
          "height": "100px",
        }
      }
    >
      <img
        alt=""
        src="http://dummyimage.com/100x100.png/ff4444/ffffff"
      />
    </div>
    <div>
      <p>
        DATE 
        2/11/2017, 1:06:40 AM
         
      </p>
      <p
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        TXT 
        Nulla tempus. Vivamus in felis eu sapien cursus vestibulum.
          
      </p>
      <span />
    </div>
  </div>
</li>
`;
